### PR TITLE
Optional ObjectStore Implementation Based on azure-storage-blob-go

### PR DESCRIPTION
**Pull request by chzar**
This PR is for an implementation of the ObjectStore interface based on the https://github.com/Azure/azure-storage-blob-go SDK which is currently in preview.

The commits I have already pushed are currently only a POC, but eventually I envision users being able to toggle between the default ObjectStore and the ObjectStorePreview using a commandline flag.

**Crucially, I intend to support client side encryption for blobs** (a feature supported by other plugins) which is not available in azure-sdk-for-go

There is an implementation for Init() and ListObjects() with unit tests.
See: https://docs.microsoft.com/en-us/azure/storage/common/storage-client-side-encryption?tabs=dotnet

Feedback is appreciated.

### CODE REVIEW

**Comment by chzar** [comment]
Setting PR to ready for review to solicit maintainer feedback.

API for the preview object storage is nearly complete. Tests are available, but will require an Azure account.

Still lots to do, but would appreciate some feedback on progress so far.

File: .gitignore [hunk 1]
```diff
@@ -17,4 +17,7 @@ debug
 *.diff
 .vs
 
-_tiltbuild
\ No newline at end of file
+_tiltbuild
+
+# Mock Configs
+mock_config_object_store_preview
```
Comment (jenting) [inline .gitignore:23]: I think we could not add this to .gitignore.

**Review by jenting — COMMENTED** [review]
To be honest, I've no knowledge of the Azure object-store preview.
Is there a GitHub issue related to this?
If possible, it'd be great if you could provide more information on the Azure object-store preview.
Thank you.

**Comment by chzar** [comment]
https://github.com/Azure/azure-storage-blob-go

The azure storage blob go is a newer Microsoft library specifically for interacting with blob storage. Crucially it supports client side encryption which is a requirement for some organizations that use cloud services.

**Comment by chzar** [comment]
I am closing this pull request. I have decided to write a separate plugin to implement this feature:  https://github.com/equinor/velero-plugin-aeos

**Pull request closed without merging**